src/Abstractions/IBlobStorage.cs

```diff
@@ -62,5 +64,16 @@ public interface IBlobStorage
         /// Retrieves metadata about every directory in the storage medium under the given path
         /// </summary>
         Task<IEnumerable<BlobItem>> GetAllDirectoriesInPathAsync(string path);
+
+        /// <summary>
+        /// Deletes blobs older than the specified cutOffDate
+        /// </summary>
+        /// <param name="cutOffDate">The DateTimeOffSet to delete blobs older than</param>
+        /// <param name="continuationToken">BlobContinuationToken</param>
+        /// <param name="cancellationToken">CancellationToken</param>
+        /// <returns></returns>
+        Task DeleteBlobsOlderAync(DateTimeOffset cutOffDate,
```

Comment:
Do not leak Azure-Specific things from this library (no BlobContinuationToken). That makes it impossible to replace with non-Azure resources and makes it untestable